src/log4stash/Authentication/AuthenticationMethodChooser.cs

```diff
@@ -9,6 +10,10 @@ public class AuthenticationMethodChooser : IAuthenticationMethodChooser
         private readonly IExternalEventWriter _eventWriter;
         private IAuthenticator _innerMethod;
 
+        public AuthenticationMethodChooser()
```

Comment:
This would break the AWS authentication method.

I think a larger refactor may be necessary here, like

1) Refactoring the `Authenticate` method to accept an `EventWriter`; or
2) Refactoring the authentication method "chooser" types to act as factories to create the appropriate authenticator when `ActivateOptions` is called on the appender.

I think (1) is probably sufficient?

Reply:
Do you think you can implement the first method?

Reply:
I'll work on that this evening.